packages/malloy-vscode/src/common/dynamic_connection_manager.ts

```diff
@@ -0,0 +1,35 @@
+import { DuckDBConnection } from "@malloydata/db-duckdb";
+import {
+  Connection,
+  FixedConnectionMap,
+  LookupConnection,
+} from "@malloydata/malloy";
+
+export class DynamicConnectionManager implements LookupConnection<Connection> {
+  private _duckdb: DuckDBConnection | null = null;
+
+  constructor(
+    private fixedConnectionMap: FixedConnectionMap,
+    private workingPath: string
+  ) {}
+
+  async lookupConnection(
+    connectionName?: string | undefined
+  ): Promise<Connection> {
+    try {
+      return await this.fixedConnectionMap.lookupConnection(connectionName);
+    } catch (err) {
+      if (connectionName === "duckdb") {
```

Comment:
Maybe instead of making a dynamic "duckdb" connection which sets the working path corresponding to the malloy file's directory, we can just make the working path of a _configured_ connection with DuckDB backend (and any name) be that directory.